### PR TITLE
4142 & 781: BAH Document type display replacement

### PR DESCRIPTION
**Pull request by robfelder**
## Description
This PR covers document type updates needed for 4142, 781 and 781a. For 4142, it will have 3 drop down options. For both 781 and 781a, they will have static text as mentioned in the copy deck.

Link to story: https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/13104
Also resolves this bug: https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/15614

## Testing done
Manual & Unit

## Screenshots
screen shot 2018-12-07 at 3 48 16 pm
screen shot 2018-12-07 at 3 48 43 pm
screen shot 2018-12-07 at 3 48 53 pm

## Acceptance criteria
- For 4142:

- [x] During the interview if the user selects to upload his/her private medical records then the Document Type' dropdown should only contain the options listed on the attached spreadsheet/tab _4142_in the VBMS Description Text column
- [x] Document name field will no longer show, it will default to the file name
- [x] User is able to delete the file if needed
- [x] User can add another document
- [x] User can continue to next step in the flow

- For 0781:

- [x] During the interview if the user specifically selects to upload a completed 0781 then the document type list will automatically default to the Form name (VA Form 21-0781-Statement in Support of Claim for PTSD)
- [x] In the scenario where the Document Type is defaulted to a specific form then it is not editable by the user
- [x] Document name field will no longer show, it will default to the file nam
- [x] User is able to delete the file if needed
- [x] User can add another document
- [x] User can continue to next step in the flow

- For 0781a:

- [x] During the interview if the user specifically selects to upload a completed 0781a then the document type list will automatically default to the Form name (VA Form 21-0781a-Statement in Support of Claim for PTSD Secondary to Personal Assault)
- [x] In the scenario where the Document Type is defaulted to a specific form then it is not editable by the user
- [x] Document name field will no longer show, it will default to the file name
- [x] User is able to delete the file if needed
- [x] User can add another document
- [x] User can continue to next step in the flow

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/pages/uploadPtsdDocuments.js [hunk 1]
```diff
@@ -1,33 +1,25 @@
...
+import fullSchema from 'vets-json-schema/dist/21-526EZ-ALLCLAIMS-schema.json';
+
+const { completedFormAttachments } = fullSchema.properties;
 
 export const uiSchema = {
   'ui:title': ptsd781NameTitle,
   'ui:description': uploadDescription,
-  ptsd781: ancillaryFormUploadUi('', 'PTSD 781 form'),
+  ptsd781: ancillaryFormUploadUi('', 'PTSD 781 form', {
+    attachmentId: 'VA Form 21-781 - Statement in Support of Claim for PTSD',
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/pages/uploadPtsdDocuments.js]: So I see the 781a is using the attachment code but here we're passing the label for the attachment code instead...
```
  {
    value: 'L229',
    label:
      'VA Form 21-0781a - Statement in Support of Claim for PTSD Secondary to Personal Assault',
  },
  {
    value: 'L228',
    label: 'VA Form 21-0781 - Statement in Support of Claim for PTSD',
  },
```
Is `L228` not valid?

File: src/applications/disability-benefits/all-claims/utils.jsx [hunk 2]
```diff
@@ -632,9 +643,13 @@ export const ancillaryFormUploadUi = (label, itemDescription) =>
     parseResponse: (response, file) => ({
       name: file.name,
       confirmationCode: response.data.attributes.guid,
+      attachmentId,
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/utils.jsx]: If this is an empty string it won't overwrite other pages that use this helper along side the documentType dropdown will it?

**Review by aaronjlech — COMMENTED** [review]
A few questions around the `attachmentId`

File: src/applications/disability-benefits/all-claims/pages/uploadPtsdDocuments.js (hunk 1 above)
Comment (robfelder) [reply src/applications/disability-benefits/all-claims/pages/uploadPtsdDocuments.js]: It's not in this case because it's a different type. Since this is of type string, it's treated differently by the system.

File: src/applications/disability-benefits/all-claims/utils.jsx (hunk 2 above)
Comment (robfelder) [reply src/applications/disability-benefits/all-claims/utils.jsx]: This seems to just set a default, or initial value. So if it's empty, and a dropdown is used, then it defaults to the empty option of the dropdown. If it's a textarea, or any other widget, it will do the same.

**Review by aaronjlech — APPROVED** [review]

File: src/platform/site-wide/sass/style-consolidated.scss [hunk 3]
```diff
@@ -42,7 +42,7 @@ $font-stack: "Source Sans Pro", "Helvetica Neue", "Helvetica", "Roboto", "Arial"
   font-family: "Source Sans Pro";
   font-style: normal;
   font-weight: 300;
-  src: url(/generated/sourcesanspro-light-webfont.eot?#iefix) format("embedded-opentype"),
+  src: url("/generated/sourcesanspro-light-webfont.eot?#iefix") format("embedded-opentype"),
```
Comment (jbalboni) [inline src/platform/site-wide/sass/style-consolidated.scss]: Can we revert the changes in here?

File: src/applications/disability-benefits/all-claims/pages/uploadPersonalPtsdDocuments.js [hunk 4]
```diff
@@ -1,33 +1,23 @@
 import { uploadDescription } from '../content/fileUploadDescriptions';
 import { ptsd781aNameTitle } from '../content/ptsdClassification';
 import { ancillaryFormUploadUi } from '../utils';
+import fullSchema from 'vets-json-schema/dist/21-526EZ-ALLCLAIMS-schema.json';
```
Comment (jbalboni) [inline src/applications/disability-benefits/all-claims/pages/uploadPersonalPtsdDocuments.js]: Is this schema in sync with the one in the vets-website repo? It seems a little concerning to use both at the same time.

File: src/platform/site-wide/sass/style-consolidated.scss (hunk 3 above)
Comment (robfelder) [reply src/platform/site-wide/sass/style-consolidated.scss]: Hey @jbalboni! Sure thing, I'll give it a try. For some reason, the linter kept screaming about it when I was trying to commit.

**Comment by robfelder** [comment]
Hey @jbalboni! I was able to revert that Sass change. Can you take another look when you get a chance?

File: src/applications/disability-benefits/all-claims/config/schema.js [hunk 5]
```diff
@@ -1170,6 +1170,48 @@ const schema = {
       type: 'array',
       items: { $ref: '#/definitions/secondaryPtsdIncident' },
     },
+    privateMedicalRecordAttachments: {
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/config/schema.js]: Do these new properties need to be added since they're already in the version of `vets-json-schema` we're using here?

**Review by jbalboni — APPROVED** [review]

File: src/applications/disability-benefits/all-claims/config/schema.js (hunk 5 above)
Comment (robfelder) [reply src/applications/disability-benefits/all-claims/config/schema.js]: Not totally. I only put it there for local testing stuff.
